src/isl/isl_range.rs

```diff
@@ -14,6 +14,81 @@ use std::cmp::Ordering;
 use std::prelude::rust_2021::TryInto;
 use std::str::FromStr;
 
+/// Provides IntegerRange type to be used to create integer ranges.
+pub type IntegerRange = RangeImpl<Integer>;
+
+impl IntegerRange {
+    pub fn new(
+        start: RangeBoundaryValue<Integer>,
+        end: RangeBoundaryValue<Integer>,
+    ) -> IonSchemaResult<IntegerRange> {
+        RangeImpl::range(start, end)
+    }
+}
+
+/// Provides IntegerNonNegativeRange type to be used to create integer non negative ranges.
+pub type IntegerNonNegativeRange = RangeImpl<usize>;
+
+/// Provides FloatRange type alias to be used to create float ranges.
+pub type FloatRange = RangeImpl<f64>;
+
+impl FloatRange {
+    pub fn new(
+        start: RangeBoundaryValue<f64>,
+        end: RangeBoundaryValue<f64>,
+    ) -> IonSchemaResult<FloatRange> {
+        RangeImpl::range(start, end)
+    }
+}
```

Comment:
As a `TODO`, I think we should make `RangeImpl::range(...)` and each of these `new(...)` methods generic over `Into<RangeBoundaryValue<_>>`.

That way, calls like this:
```rust
FloatRange::new(
   2e1.into(),
   5e1.into()
)
```
can be reduced to:
```rust
FloatRange::new(2e1, 5e1)
```

Reply:
Yes, I will add this change.